### ISSUE TITLE
Log a message when a resource is ignored from patching by Zarf Agent

### ISSUE THREAD

**Issue by andre-aktivconsultancy** [defenseunicorns/zarf#1744]
### Is your feature request related to a problem? Please describe.
Today I deployed some images and a Helm chart with Zarf. The pods all ran into image pull errors. The PodSpecs of my pods where not updated by the Zarf Agent. It took me a long time to figure out that this was caused by a namespace that existed before `zarf init` was executed. The namespace got labeled with `zarf.dev/agent: ignore`. A debug log from the Zarf Agent stating that it ignored patching some resources, because of the namespace being ignored would have saved a lot of time.

I do realize that behavior is as intended and is documented in the FAQ.

### Describe the solution you'd like

- **When** a resource being ignored from patching by Zarf Agent
- **Then** log a debug message explaining why the resource is ignored

**Comment by Racer159** [comment]
This may be difficult to add in cluster because we actually don't receive the initial webhook traffic due to our webhook `matchExpressions` configuration excluding those labels.

If we added a warning when we saw an existing namespace that had an ignore label on it during the deployment run would that work? We could add logic in/around this if block to do that check https://github.com/defenseunicorns/zarf/blob/main/src/internal/packager/helm/post-render.go#L198

**Comment by andre-aktivconsultancy** [comment]
Yeah I guess that would work as well, when inspecting the zarf-agent logs it would show up and it would have been the hint to point me in the right direction quicker :)
